lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -485,25 +487,36 @@ impl DomLayers {
         let front = DomScene::new(logger);
         let fullscreen_vis = DomScene::new(logger);
         let back = DomScene::new(logger);
+        let panel = DomScene::new(logger);
+
+        panel.dom.set_class_name("panel");
+        panel.dom.set_style_or_warn("pointer-events", "auto", logger);
+        panel.dom.set_style_or_warn("z-index", "0", logger);
+
+        back.dom.set_class_name("back");
+        back.dom.set_style_or_warn("pointer-events", "auto", logger);
+        back.dom.set_style_or_warn("z-index", "1", logger);
+
+        fullscreen_vis.dom.set_class_name("fullscreen_vis");
+        fullscreen_vis.dom.set_style_or_warn("z-index", "2", logger);
+
         canvas.set_style_or_warn("height", "100vh", logger);
         canvas.set_style_or_warn("width", "100vw", logger);
         canvas.set_style_or_warn("display", "block", logger);
         // Position must not be "static" to have z-index working.
         canvas.set_style_or_warn("position", "absolute", logger);
-        canvas.set_style_or_warn("z-index", "2", logger);
+        canvas.set_style_or_warn("z-index", "3", logger);
         canvas.set_style_or_warn("pointer-events", "none", logger);
+
         front.dom.set_class_name("front");
-        front.dom.set_style_or_warn("z-index", "3", logger);
-        back.dom.set_class_name("back");
-        back.dom.set_style_or_warn("pointer-events", "auto", logger);
-        back.dom.set_style_or_warn("z-index", "0", logger);
-        fullscreen_vis.dom.set_class_name("fullscreen_vis");
-        fullscreen_vis.dom.set_style_or_warn("z-index", "1", logger);
-        dom.append_or_panic(&canvas);
-        dom.append_or_panic(&front.dom);
+        front.dom.set_style_or_warn("z-index", "4", logger);
+
+        dom.append_or_panic(&panel.dom);
         dom.append_or_panic(&back.dom);
         dom.append_or_panic(&fullscreen_vis.dom);
-        Self { back, fullscreen_vis, front, canvas }
+        dom.append_or_panic(&canvas);
+        dom.append_or_panic(&front.dom);
+        Self { back, panel, fullscreen_vis, front, canvas }
```

Comment:
I like the division into "sections" with newlines in this function. It used to look OK before we used autoformatter, but such cleaning is needed, nice job. However, I don't understand why some of the things are sorted and some not – like you have a group of lines where you are creating all of the objects, then you have groups where you are setting up parameters per layer, and then you have a group of lines where you add all layers to the parent layer. If we want to group the lines by layer, lets be consistent.

Reply:
I feel this is just a matter of personal taste. I don't dislike `created all, set up 1, set up 2, ..., finish all` ordering, but I changed as you asked.